fdbrpc/Base64Decode.cpp

```diff
@@ -0,0 +1,419 @@
+/*
+ * Base64Decode.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string_view>
+#include <fmt/format.h>
+#include "fdbrpc/Base64Encode.h"
+#include "fdbrpc/Base64Decode.h"
+#include "flow/flow.h"
+#include "flow/Arena.h"
+#include "flow/Error.h"
+#include "flow/UnitTest.h"
+
+namespace {
+
+constexpr uint8_t _X = 0xff;
+
+template <bool UrlDecode>
+inline uint8_t decodeValue(uint8_t valueIn) noexcept {
```

Comment:
Please add a comment here to describe how you built these lookup tables

Reply:
Done!